SHARE-247 Refactoring - Media Package Categories

FelixAuf opened the pull request:
Added ID and Priority columns for consistency. Added a check for deleting categories, if they are in use by a media package file the site used to crash with an exception, now it shows an error message. Added tests.

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

dmetzner reviewed (changes requested):
Feature looks good, please just fix the test

FelixAuf committed:
SHARE-247 Refactoring - Media Package Categories

Added ID and Priority columns for consistency. Added a check for deleting categories, if they are in use by a media package file the site used to crash with an exception, now it shows an error message. Added tests.

FelixAuf commented:
> Feature looks good, please just fix the test

@dmetzner Done, don't know why the test failed though as the error was already moved in that commit
